From 2b055cbfcff84002a4f71eb4476e7b0c2ec462ef Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Thu, 15 Oct 2020 16:56:54 +0700
Subject: [PATCH 283/446] ARM: dts: r8a7742: Add TPU/PWM device nodes

Add tpu and pwm[0-6] nodes to the Renesas R8A7742 (RZ/G1H) SoC dtsi.

Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 73 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 73 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 870872d..e94cdcf 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1276,6 +1276,79 @@
 			status = "disabled";
 		};
 
+		tpu: pwm@e60f0000 {
+			compatible = "renesas,tpu-r8a7742", "renesas,tpu";
+			reg = <0 0xe60f0000 0 0x40>;
+			interrupts = <0 135 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp3_clks R8A7742_CLK_TPU0>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+			#pwm-cells = <3>;
+		};
+
+		pwm0: pwm@e6e30000 {
+			compatible = "renesas,pwm-rcar";
+			reg = <0 0xe6e30000 0 0x8>;
+			clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+			#pwm-cells = <2>;
+		};
+
+		pwm1: pwm@e6e31000 {
+			compatible = "renesas,pwm-rcar";
+			reg = <0 0xe6e31000 0 0x8>;
+			clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+			#pwm-cells = <2>;
+		};
+
+		pwm2: pwm@e6e32000 {
+			compatible = "renesas,pwm-rcar";
+			reg = <0 0xe6e32000 0 0x8>;
+			clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+			#pwm-cells = <2>;
+		};
+
+		pwm3: pwm@e6e33000 {
+			compatible = "renesas,pwm-rcar";
+			reg = <0 0xe6e33000 0 0x8>;
+			clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+			#pwm-cells = <2>;
+		};
+
+		pwm4: pwm@e6e34000 {
+			compatible = "renesas,pwm-rcar";
+			reg = <0 0xe6e34000 0 0x8>;
+			clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+			#pwm-cells = <2>;
+		};
+
+		pwm5: pwm@e6e35000 {
+			compatible = "renesas,pwm-rcar";
+			reg = <0 0xe6e35000 0 0x8>;
+			clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+			#pwm-cells = <2>;
+		};
+
+		pwm6: pwm@e6e36000 {
+			compatible = "renesas,pwm-rcar";
+			reg = <0 0xe6e36000 0 0x8>;
+			clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+			#pwm-cells = <2>;
+		};
+
 		scifb0: serial@e6c20000 {
 			compatible = "renesas,scifb-r8a7742",
 				     "renesas,scifb";
-- 
2.7.4

